ghproxy/ghcache/ghcache.go

```diff
@@ -202,39 +211,55 @@ func (u upstreamTransport) RoundTrip(req *http.Request) (*http.Response, error)
 // NewDiskCache creates a GitHub cache RoundTripper that is backed by a disk
 // cache.
 func NewDiskCache(delegate http.RoundTripper, cacheDir string, cacheSizeGB, maxConcurrency int) http.RoundTripper {
-	return NewFromCache(delegate, diskcache.NewWithDiskv(
-		diskv.New(diskv.Options{
-			BasePath:     path.Join(cacheDir, "data"),
-			TempDir:      path.Join(cacheDir, "temp"),
-			CacheSizeMax: uint64(cacheSizeGB) * uint64(1000000000), // convert G to B
-		})),
+	return NewFromCache(delegate,
+		func(partitionKey string) httpcache.Cache {
+			return diskcache.NewWithDiskv(
+				diskv.New(diskv.Options{
+					BasePath:     path.Join(cacheDir, "data", partitionKey),
+					TempDir:      path.Join(cacheDir, "temp", partitionKey),
+					CacheSizeMax: uint64(cacheSizeGB) * uint64(1000000000), // convert G to B
```

Comment:
This makes the `cacheSizeGB` misleading as it's the size of one cache partition and not the whole thing, right?

Reply:
yes, thats why I updated the flag description